array.go

```diff
@@ -913,3 +913,59 @@ func countElement(array []interface{}, element interface{}) int {
 	}
 	return count
 }
+
+// Transform runs the passed function on all the Elements in the array.
+//
+// If transformation inside function fails, the original Array is marked failed.
+//
+// Transform will immediately stop execution of the function
+// upon first transformation failures
+// or if return value is nil.
+//
+// Example:
+//
+//	array := NewArray(t, []interface{}{"foo", "bar"})
+//	transformedArray := array.Transform(
+//		func(index int, value *httpexpect.Value) *httpexpect.Value {
+//			return *httpexpect.NewValue(...)
+//		},
+//	)
+func (a *Array) Transform(fn func(index int, value *Value) *Value) *Array {
+	a.chain.enter("Transform()")
+	defer a.chain.leave()
+
+	if a.chain.failed() {
+		return a
+	}
```

Comment:
Let's be consistent and ensure that Transform() always returns a new array, no matter if it failed or succeeded:

```
return newArray(a.chain, nil)
```

Same in other places.